TohLiYuan iP

TohLiYuan opened the pull request:
# KnowledgeYuan

> l have a brain like a sieve - it's constantly leaking memories.

If you ever had that experience, then KnowledgeYuan is for you!
KnowledgeYuan *frees* your **brain capacity** from ever requiring to remember task and deadlines in the future!

Features:
- Create reminders for upcoming activities
- Mark reminders as complete/incomplete
- List all available reminders
- Find reminders with a given keyword

Installation:
1. Download the binary from [release](https://github.com/TohLiYuan/ip/releases/tag/A-Jar)
2. Run the jar file 
```
java -jar duke.jar
```
3. Enjoy! 😏

Developmental Plans:
- [x] Find tasks with given keyword
- [ ] Better parser
- [ ] Support for more robust DateTime system

If you want to launch the application in Java, this is the main method:
```Java
public static void main(String[] args) {
    Application.launch(Duke.class, args);
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

GamerYuan committed:
Rename & Skeletal Version

GamerYuan committed:
Refactor, Chat Echoing

GamerYuan committed:
Message Records & Command Refactor

GamerYuan committed:
Tasks & Marking, renamed variables to follow convention

GamerYuan committed:
Task Types, Refactor

GamerYuan committed:
Minor Formatting

GamerYuan committed:
Automated Test

GamerYuan committed:
Error Handling

GamerYuan committed:
Deleting Tasks

GamerYuan committed:
Primitive saving mechanism (to be tested)

GamerYuan committed:
Refactored new saving logic, better implementation

GamerYuan committed:
Merge branch 'branch-level-7'

GamerYuan committed:
Updated gitignore, cleanup

GamerYuan committed:
Class Refactor

GamerYuan committed:
Primitive DateTime for deadline and event

GamerYuan committed:
Merge branch 'branch-level-8'

GamerYuan committed:
Commands

GamerYuan committed:
Moved commands to default package

GamerYuan committed:
Command Parser

GamerYuan committed:
Better Save Logic

GamerYuan committed:
Implement UI

GamerYuan committed:
Better regex matching

GamerYuan committed:
Merge branch 'A-MoreOOP'

GamerYuan committed:
Organise packages

GamerYuan committed:
Merge branch 'branch-A-Packages'

GamerYuan committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

GamerYuan committed:
Add Gradle Support

GamerYuan committed:
Add JUnit tests

GamerYuan committed:
Merge branch 'branch-A-JUnit'

GamerYuan committed:
Configure jar build

GamerYuan committed:
Fix empty command parse bug

GamerYuan committed:
Add JavaDocs

GamerYuan committed:
Follow coding standard

GamerYuan committed:
Add more JavaDocs

GamerYuan committed:
Add Find command

GamerYuan committed:
Merge tag 'A-JavaDoc'

JavaDoc

GamerYuan committed:
Resolve merge conflict for branch-A-CodingStandard

GamerYuan committed:
Resolve merge conflict for branch-Level-9

GamerYuan committed:
Checkstyle fixes

junhonglow commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,44 @@
+public abstract class Task {
+    protected static final String DISCRIMINATOR = " || ";
+    protected String name;
+    protected boolean status;
```
Perhaps you could use a name to make the status sound more like a boolean.

TohLiYuan replied:
Good point, noted for future revisions

junhonglow reviewed (commented):
Overall, the coding standards were well followed, with a small nitpick mentioned.
I like that the javadocs were included for the public methods and classes, and methods were also generally descriptive of their function.

GamerYuan committed:
Add basic UI and custom dialog box

GamerYuan committed:
Add image clipping

GamerYuan committed:
Implement fxml

GamerYuan committed:
Add responsive UI

GamerYuan committed:
Merge branch 'branch-Level-10'

GamerYuan committed:
Use Gradle CheckStyle

GamerYuan committed:
Merge branch 'branch-A-CheckStyle'

Sheeepen commented on src/main/java/duke/commands/InvalidCommand.java:
I like the use of commands! Maybe one helpful Command to add would be a HelpCommand (possibly with the "help" keyword) that lists all possible actions the user can take.

Sheeepen commented on src/main/java/duke/parser/CommandParser.java:
```diff
@@ -0,0 +1,89 @@
+ * @author Toh Li Yuan (A0255811H)
+ */
+public class CommandParser {
+    private static final Pattern COMMAND_PATTERN = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+
+    private static final Pattern TODO_ARGS_PATTERN = Pattern.compile("(?<name>\\S.*)");
+    private static final Pattern DEADLINE_ARGS_PATTERN = Pattern.compile("(?<name>\\S+.*)( /by )(?<time>\\S.*)");
+    private static final Pattern EVENT_ARGS_PATTERN =
```
Great effort in using Pattern and Regex classes in your Parser classes!

Sheeepen commented on src/main/java/duke/records/ChatRecord.java:
```diff
@@ -0,0 +1,105 @@
+ * Records all tasks.
+ * @author Toh Li Yuan (A0255811H)
+ */
+public class ChatRecord {
```
I personally think perhaps this class could be named better to reflect its functionality. (Personally I use TaskList)

Sheeepen commented on src/main/java/duke/task/TaskTypes.java:
```diff
@@ -0,0 +1,11 @@
+ * Contains all available Task Types.
+ * @author Toh Li Yuan (A0255811H)
+ */
+public enum TaskTypes {
```
Nice usage of enums that allows for easy expansion of future TaskTypes!

Sheeepen commented on src/test/java/duke/parser/CommandParserTest.java:
```diff
@@ -0,0 +1,66 @@
+
+import duke.commands.Command;
+
+public class CommandParserTest {
```
Nice tests implemented. Perhaps for completion you could implement the rest of the tests as well.

Sheeepen commented on src/test/java/duke/parser/CommandParserTest.java:
```diff
@@ -0,0 +1,66 @@
+        assertEquals(testInput, cmd.toString());
+    }
+
+    @Test
+    public void invalidParseTest() {
+        CommandParser cp = new CommandParser();
+        String testInput = "test123 test456";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals("Invalid Command! Command not found", cmd.toString());
+    }
+
+    @Test
+    public void invalidParseTest2() {
+        CommandParser cp = new CommandParser();
+        String testInput = "todo";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals("Invalid Command! Name cannot be empty!", cmd.toString());
+    }
+
+    @Test
+    public void complexParseTest() {
+        CommandParser cp = new CommandParser();
+        String testInput = "todo deadline event /from 8/2/2023 1200 /to 8/3/2023 1400 /to 6/5/2023";
+        Command cmd = cp.parseCommand(testInput);
+        assertNotNull(cmd);
+        assertEquals(testInput, cmd.toString());
+    }
```
Perhaps you could have better names for these tests? Or add descriptions as to _how/why_ they are invalid. Maybe nesting these tests under one invalidTest could work too.

Sheeepen reviewed (commented):
LGTM. Mostly just minor nits to fix and completing test cases.

ceilingFan456 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,70 @@
+        return out;
+    }
+
+    public boolean getTermination() {
```
Should the name be modified slightly to make it sound like it returns a boolean?

ceilingFan456 commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,13 @@
+ * The abstract class for commands.
+ * @author Toh Li Yuan (A0255811H)
+ */
+public abstract class Command {
```
I really like the use of abstract class here.

ceilingFan456 commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+ * @author Toh Li Yuan (A0255811H)
+ */
+public class FindCommand extends Command {
+    public static final String COMMAND_PHRASE = "find";
```
I like the use of constants here.

ceilingFan456 commented on src/main/java/duke/parser/CommandParser.java:
```diff
@@ -0,0 +1,89 @@
+        String args = matcher.group("arguments");
+
+        switch (commandWord) {
+        case MarkCommand.COMMAND_PHRASE:
```
you have followed the switch statement format nicely.

ceilingFan456 commented on src/main/java/duke/parser/TaskParser.java:
```diff
@@ -0,0 +1,65 @@
+     * @param tasks the array of Task to be stored.
+     * @return the formatted string to be stored.
+     */
+    public static String formatSave(Task[] tasks) {
```
perhaps the name can be modified to make it sound like it returns a String. It kind of sounds like the return type is void.

ceilingFan456 commented on src/test/java/duke/parser/CommandParserTest.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    @Test
+    public void invalidParseTest2() {
```
Perhaps it might be good to leave a bit of comment for the invalidParseTest2 to show the difference with invalidParseTest.

ceilingFan456 reviewed (commented):
Really good job on following the naming conventions and coding standards. Your code is clear and clean to read.

GamerYuan committed:
Change method names

The following method names have been changed to sound more intuitive as per their return types:
- Duke::getTermination() -> Duke::isTerminate()
- TaskParser::formatSave(Task[]) -> TaskParser::generateSaveString(Task[])

Other changes:
- Update build.gradle for JavaFX JAR file
- Update Duke.java to exclude Duke::main(String[]) in production code

GamerYuan committed:
Use Assertions

Assertions are added to assume values that the program require to run smoothly, to prevent unexpected errors due to unchecked code

Other changes:
- Optimised the shutdown logic when typing the bye command

GamerYuan committed:
Remove magic values

Remove hard code values
- Use Task.DISCRIMINATOR instead of '||' in TaskParser::parseSave()

Commit:
Merge pull request #1 from TohLiYuan/branch-A-Assertions

Use Assertions

Commit:
Merge pull request #2 from TohLiYuan/branch-A-CodeQuality

Remove magic values

GamerYuan committed:
Add reminders

Reminders can remind the user when a task is about to reach or has reached its deadline or ending date

New Functionality: Reminds the user of urgent tasks ending in a day upon app launch

New Command: Remind Command
Remind Command allows users to specify the scope of days to seek for tasks

eg. remind 10
This command will show the tasks reaching its deadline within 10 days

If no number of days is specified, the command defaults itself to 7 days

GamerYuan committed:
Merge branch 'branch-B-Reminders'

GamerYuan committed:
Cleanup code and style checks

GamerYuan committed:
Adjust main window UI

GamerYuan committed:
Add window title

GamerYuan committed:
Add ui screenshot

GamerYuan committed:
Change reminder feature logic

GamerYuan committed:
Add UG

GamerYuan committed:
Minor changes

GamerYuan committed:
Add TaskParser Tests

GamerYuan committed:
Merge branch 'branch-A-MoreTesting'

GamerYuan committed:
Follow stylechecks for new tests

GamerYuan committed:
Merge branch 'branch-A-MoreTesting'

GamerYuan committed:
Change method names

GamerYuan committed:
Fix JavaFX Linux issue
